stor/tests/test_integration_dx.py

```diff
@@ -0,0 +1,320 @@
+import gzip
+import os
+import pytest
+import time
+import unittest
+from unittest import skipIf
+
+import dxpy
+import mock
+import six
+
+from stor import exceptions
+from stor import NamedTemporaryDirectory
+from stor import Path
+from stor.dx import DXPath
+from stor.test import DXTestCase
+from stor.tests.shared import assert_same_data
+from stor.tests.test_integration import BaseIntegrationTest
+import stor
+import stor.tests.test_integration as test_integration
+
+
+class DXIntegrationTest(BaseIntegrationTest.BaseTestCases, DXTestCase):
+    def setUp(self):
+        self.vcr_enabled = False  # don't want vcr playback for integration tests
+        super(DXIntegrationTest, self).setUp()
+        if not os.environ.get('DX_AUTH_TOKEN'):
+            raise unittest.SkipTest(
+                'DX_AUTH_TOKEN env var not set. Skipping integration test')
+
+        self.setup_temporary_project()
+        self.project = DXPath('dx://' + self.project + ':')
+        self.test_dir = self.project / 'test'
+
+    def tearDown(self):
+        super(DXIntegrationTest, self).tearDown()
+
+    def test_login_auth(self):
+        def mock_header(r, security_context):
+            auth_header = security_context["auth_token_type"] + " " + \
+                          security_context["auth_token"]
+            r.headers[b'Authorization'] = auth_header.encode()
+            return r
+        # dxpy.AUTH_HELPER gets set upon login and called with each api which we mock out here
+        with mock.patch('dxpy.AUTH_HELPER', autospec=True) as mock_auth:
+            mock_auth.security_context = {
+                'auth_token_type': 'Bearer',
+                'auth_token': 'PUBLIC'
+            }
+            mock_auth.side_effect = lambda x: mock_header(x, mock_auth.security_context)
+            with pytest.raises(exceptions.NotFoundError, match='no projects'):
+                self.test_dir.makedirs_p()
+            self.assertEqual(mock_auth.call_count, 1)
+        self.test_dir.makedirs_p()
+        self.assertTrue(self.test_dir.isdir())
+
+    def test_read_bytes_from_binary(self):
+        test_file = self.test_dir / 'test_file.txt'
+        with stor.open(test_file, mode='wb') as fp:
+            fp.write(test_integration.BYTE_STRING)
+        file_h = dxpy.DXFile(dxid=test_file.canonical_resource,
+                             project=test_file.canonical_project)
+        file_h.wait_on_close(20)  # wait for file to go to closed state
+        with stor.open(test_file, mode='rb') as fp:
+            result = fp.read()
+        assert result == test_integration.BYTE_STRING
+
+    def test_copy_to_from_dir(self):
```

Comment:
why can't test on subclass work?

Reply:
I am guessing you meant the same as above (why not inherit from super class)? If yes, see above. If not, let me know what you mean.

Reply:
Fixed.